src/server/plugins/initialize.js

```diff
@@ -40,6 +40,10 @@ module.exports = async function (kbnServer, server, config) {
     path.pop();
   };
 
+  for (let plugin of plugins) {
+    await plugin.preInit();
```

Comment:
Generally speaking, I'm against implementing something we don't need for a current use case ([speculative generality is a code smell](https://sourcemaking.com/refactoring/smells/speculative-generality)). In this case, however, implementing pre-initialization to parallel initialization would lend consistency to the codebase, and make it easier to reason about IMO.